### ISSUE TITLE
What's the difference between BatchNormalization mode=1 and Layer Normalization?

### ISSUE THREAD

**Issue by Kyubyong** [fchollet/keras#3756]
To me, layer normalization (https://arxiv.org/pdf/1607.06450v1.pdf) looks the same as the sample-wise batch normalization in Keras BatchNormalization mode=1. What's their difference?

**Comment by MartinThoma** [comment]
It seems as if mode was removed in Keras 2.0: https://keras.io/layers/normalization/ ? (By the way: Is there a migration guide?)

**Comment by rkoppula** [comment]
Yes, some documentation with examples to migrate BatchNormalization layers will be helpful.

**Comment by rkoppula** [comment]
I found this in the 2.0 release notes - "(use batch metrics for feature-wise normalization during training, and use moving metrics for feature-wise normalization during testing)." 

If I am interpreting it right, it is not possible to get the old behavior. Is that right?

**Comment by Irtza** [comment]
I can't seem to replicate behavior for Batch-normalization Layer mode=2 , in keras 2. Can someone help? @rkoppula , @Kyubyong  @MartinThoma .

**Comment by HitLuca** [comment]
@Irtza 

Referencing [this solution](https://www.bountysource.com/issues/43244022-big-problems-with-gan-and-keras-2-0-because-of-new-batchnorm) the suggested way for fixing the new BatchNormalization is to first copy the keras class into another new custom class and then change this

```
return K.in_train_phase(normed_training,
                                normalize_inference,
                                training=training)
```

to this

```
return K.in_train_phase(normed_training,
                                normalize_inference,
                                training=True)
```

from my own experiments (WGAN-GP on MNIST), the new implementation seems to work. Let me know if it solved your problem.

**Comment by deepitapai** [comment]
@HitLuca 

Does this mean that I don't need to specify mode=2 while calling this custom BatchNormalization2() ?

**Comment by HitLuca** [comment]
@deepitapai yes, in this case you don't have to. Please take note that it may not have the behavior you need, it works for GAN training at least. This workaround is needed when using the newer Keras versions, as the mode parameter has been removed.
